### ISSUE TITLE
what is happening..?

### ISSUE THREAD

**Issue by zzhangyue** [nus-cs2030/2223-s2#209]
# Summary

I need some help :")
Thank you :")

## Description:

Basically I am stuck at this test case & I was wondering if it has to do with how I defined my abstract classes...? (I made Driver for NormalCab & PrivateCar, Service for JustRide/TakeACab/ShareARide)

## Screenshots

<img width="1174" alt="Screenshot 2023-02-21 at 23 16 07" src="https://user-images.githubusercontent.com/122248591/220386671-e1510617-b3d4-4a0c-976d-96f7b70d5e9a.png">

**Comment by ringleading** [comment]
could it be a toString() issue? seems like the Service type of the booking got hardcoded somewhere?

**Comment by zzhangyue** [comment]
helloooo I have attached my code snippet... would appreciate some help :"""""")
<img width="316" alt="Screenshot 2023-02-21 at 23 44 32" src="https://user-images.githubusercontent.com/122248591/220392219-8a2381b0-0418-4651-b439-50f42f8a920a.png">

**Comment by ringleading** [comment]
can u post one of ur booking's toString() instead? maybe your handling of the Service's name went off somewhere when finding the cheapest one?

**Comment by xZealthiusx** [comment]
https://github.com/nus-cs2030/2223-s2/issues/177
Can refer to this as well if it helps :)

**Comment by jeffkhong11** [comment]
> #177 Can refer to this as well if it helps :)

Yes, you have to make sure that your method of calculating fares is general enough for all drivers and services besides the ones in the question itself

**Comment by kyledilao777** [comment]
did u add a new line to the string generated?

**Comment by JYeoZS** [comment]
Legit look like typo. Problem may be in your booking class (not service class), how did you print that line out?

**Comment by clarwen** [comment]
I had this issue as well. The problem lies with Driver being not general enough. Turns out that my immutable list of possible services for Driver is in my Driver class itself. Instead, I have to delegate these services into my Driver subclass instead.

**Comment by littlemidget3** [comment]
looks like it could be a small error in your toString method, could try looking at that. Apologies for the late comment
